clouddriver-kubernetes/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/caching/KubernetesV2ProviderSynchronizable.java

```diff
@@ -196,14 +195,17 @@ private void synchronizeKubernetesV2Provider(
         log.info("Adding {} agents for account {}", newlyAddedAgents.size(), credentials.getName());
 
         kubernetesV2Provider.stageAllAgents(newlyAddedAgents);
+        stagedAccountNames.add(credentials.getName());
+      } catch (Exception e) {
+        log.warn(
+            "Error encountered scheduling new agents for account {} -- using old agent set instead",
```

Comment:
In this case we're still using the old agent set of the failed account because `ProviderUtils.unscheduleAndDeregisterAgents` will not include the failed account name. So in practice the changed account is still being preiodically cached (and force cache refresh still works for it) if its new agent set fails to build.